third_party/aws/workspace.bzl

```diff
@@ -9,11 +9,11 @@ def repo():
     third_party_http_archive(
         name = "aws",
         urls = [
-            "https://mirror.bazel.build/github.com/aws/aws-sdk-cpp/archive/1.7.266.tar.gz",
-            "https://github.com/aws/aws-sdk-cpp/archive/1.7.266.tar.gz",
-        ],
-        sha256 = "39fd8a2999260d2b8fcbc8187f1ed5299972c2b8bd14adb7850fd674fea67fb7",
-        strip_prefix = "aws-sdk-cpp-1.7.266",
+             "https://mirror.bazel.build/github.com/aws/aws-sdk-cpp/archive/1.7.336.tar.gz",
+             "https://github.com/aws/aws-sdk-cpp/archive/1.7.336.tar.gz",
+         ],
+        sha256 = "758174f9788fed6cc1e266bcecb20bf738bd5ef1c3d646131c9ed15c2d6c5720",
+        strip_prefix = "aws-sdk-cpp-1.7.336",
```

Comment:
Let's separate this into a different PR since it's an orthogonal change.

Reply:
Actually the rest of the PR depends on this new version as I added a change in the SDK to enable download of a given range of file instead of the whole file only.

Reply:
Makes sense.

Was thinking that we can add the SDK change first and then the rest of the PR. If the SDK causes internal failures we can roll that back easier.

But it's ok to keep as it is then.